lib/mayaUsd/ufe/private/UfeNotifGuard.h

```diff
@@ -89,6 +89,29 @@ class InTransform3dChange
     static bool inTransform3dChange();
 };
 
+//! \brief Helper class to scope when we are in an attribute metadata change operation.
+//
+// This simple guard class can be used within a single scope, but does not have
+// recursive scope capability.
+class InAttributeMetadataChange
+{
+public:
+    InAttributeMetadataChange() { inGuard = true; }
```

Comment:
Does not support two levels of guard active at the same time. Please use a counter instead.